src/main/java/org/jenkinsci/plugins/mesos/MesosSlaveInfo.java

```diff
@@ -0,0 +1,186 @@
+
+    private final String value;
+    private final boolean executable;
+    private final boolean extract;
+
+    public URI(String value, boolean executable, boolean extract) {
+      this.value = value;
+      this.executable = executable;
+      this.extract = extract;
+    }
+
+    public String getValue() {
+      return value;
+    }
+  }
+
+  public static class ContainerInfo {
+
+    private final String type;
+    private final String dockerImage;
+    private final List<Volume> volumes;
+    private final List<Parameter> parameters;
+    private final String networking;
+    private static final String DEFAULT_NETWORKING = Network.BRIDGE.name();
+    private final List<PortMapping> portMappings;
+    private final List<NetworkInfo> networkInfos;
+    private final boolean useCustomDockerCommandShell;
+    private final String customDockerCommandShell;
+    private final boolean dockerPrivilegedMode;
+    private final boolean dockerForcePullImage;
+    private final boolean dockerImageCustomizable;
+
+    private ContainerInfo(
+        String type,
+        String dockerImage,
+        boolean dockerPrivilegedMode,
+        boolean dockerForcePullImage,
+        boolean dockerImageCustomizable,
+        boolean useCustomDockerCommandShell,
+        String customDockerCommandShell,
+        List<Volume> volumes,
+        List<Parameter> parameters,
+        String networking,
+        List<PortMapping> portMappings,
+        List<NetworkInfo> networkInfos) {
+      this.type = type;
+      this.dockerImage = dockerImage;
+      this.dockerPrivilegedMode = dockerPrivilegedMode;
+      this.dockerForcePullImage = dockerForcePullImage;
+      this.dockerImageCustomizable = dockerImageCustomizable;
+      this.useCustomDockerCommandShell = useCustomDockerCommandShell;
+      this.customDockerCommandShell = customDockerCommandShell;
+      this.volumes = volumes;
+      this.parameters = parameters;
+      this.networkInfos = networkInfos;
+
+      if (networking == null) {
+        this.networking = DEFAULT_NETWORKING;
+      } else {
+        this.networking = networking;
+      }
+
+      if (Network.HOST.equals(Network.valueOf(networking))) {
+        this.portMappings = Collections.emptyList();
+      } else {
+        this.portMappings = portMappings;
+      }
+    }
+  }
+
+  public static class Parameter {
+
+    private final String key;
+    private final String value;
+
+    public Parameter(String key, String value) {
+      this.key = key;
+      this.value = value;
+    }
+  }
+
+  static class Volume {
+
+    private final String containerPath;
+    private final String hostPath;
+    private final boolean readOnly;
+
+    private Volume(String containerPath, String hostPath, boolean readOnly) {
+      this.containerPath = containerPath;
+      this.hostPath = hostPath;
+      this.readOnly = readOnly;
+    }
+  }
+
+  static class PortMapping {
+
+    // TODO validate 1 to 65535
+    private final Integer containerPort;
```

Comment:
Why Integer-Objects? Can those fields be NULL?

Reply:
This is a copy again. I think they can be `null`.